### PR TITLE
[SPARK-35606][PYTHON][INFRA] List Python 3.9 installed libraries in build_and_test workflow

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?

In the build_and_test workflow, tests are run against both Python 3.6 and Python 3.9. However, only libraries installed in Python 3.6 are listed. We should list Python 3.9's installed libraries as well.

### Why are the changes needed?

Listing Python 3.9's installed libraries is helpful for debugging.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual check.

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -217,6 +217,11 @@ jobs:
       run: |
         python3.6 -m pip install numpy 'pyarrow<3.0.0' pandas scipy xmlrunner plotly>=4.8
         python3.6 -m pip list
+    - name: List Python packages (Python 3.9)
+      run: |
+        curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+        python3.9 get-pip.py
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:223]: I think we can assume that `pip` is already installed.  just `python3.9 -m pip list` should work.

**Comment by HyukjinKwon** [comment]
cc @dongjoon-hyun FYI. The purpose of this PR is to list up the installed depenedncies of Python 3.9 in CI builds.

**Comment by dongjoon-hyun** [comment]
Yep. Thank you for pinging me, @HyukjinKwon .

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (xinrong-databricks) [reply .github/workflows/build_and_test.yml]: Makes sense, updated!

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -217,6 +217,9 @@ jobs:
       run: |
         python3.6 -m pip install numpy 'pyarrow<3.0.0' pandas scipy xmlrunner plotly>=4.8
         python3.6 -m pip list
+    - name: List Python packages (Python 3.9)
+      run: |
+        python3.9 -m pip list
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:222]: oh, interesting .. https://github.com/xinrong-databricks/spark/runs/2724401328?check_suite_focus=true

```
/usr/bin/python3.9: No module named pip
```

it complains that `python3.9` doesn't have `pip` .. but assuming the skip tests checks (https://github.com/apache/spark/runs/2707956326 for https://github.com/apache/spark/commit/c225196be0d18975a3d290b0b9d7283d764be322), seems all packages are installed properly.

Comment (HyukjinKwon) [reply .github/workflows/build_and_test.yml:222]: @dongjoon-hyun did you use conda to install these packages in the docker image?

Comment (HyukjinKwon) [reply .github/workflows/build_and_test.yml:222]: reading the update for docker image (https://github.com/dongjoon-hyun/ApacheSparkGitHubActionImage/commit/e6e1d1a62e1db1b3db878a16002a4704f2c65535) seems like it should have `pip` ... weird ..

Comment (xinrong-databricks) [reply .github/workflows/build_and_test.yml:222]: I see what you mean. Would you think we can afford to install pip again? It takes ~5s including listing the packages.

Comment (xinrong-databricks) [reply .github/workflows/build_and_test.yml:222]: CC @dongjoon-hyun Do you have any insights?

Comment (dongjoon-hyun) [reply .github/workflows/build_and_test.yml:222]: Thank you for pinging me, @xinrong-databricks . Let me check for you.

Comment (dongjoon-hyun) [reply .github/workflows/build_and_test.yml:222]: I confirmed the issue. We are having 3 python versions and installed in the following sequence.
- Python 3.8
- Python 3.9
- PyPy 3.7

The last thing do and it seems to break something.
```
curl -sS https://bootstrap.pypa.io/get-pip.py | pypy3
```

Let me update the image once more. And, I'll add `plotly` to Python 3.8, too.

**Comment by dongjoon-hyun** [comment]
BTW, @xinrong-databricks and @HyukjinKwon .

Although I'm adding pip explicitly, as a side note, we don't need to check this inside GitHub Action. It's a docker image.
```
$ docker run -it --rm dongjoon/apache-spark-github-action-image:20210530 python3.8 -c 'import numpy'
$ docker run -it --rm dongjoon/apache-spark-github-action-image:20210530 python3.8 -c 'import plotly'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'plotly'
```

The other packages are installed correctly. Only `pip` seems to be screwed for some reason during `pypy3 pip` installation.

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (xinrong-databricks) [reply .github/workflows/build_and_test.yml:222]: Thanks @dongjoon-hyun for help! I will put the PR on hold then.

**Comment by dongjoon-hyun** [comment]
The image is ready. I'll make a PR to update it and ping you.
```
$ docker run -it --rm dongjoon/apache-spark-github-action-image:20210602 python3.8 -m pip list
Package             Version
------------------- --------------------
certifi             2019.11.28
chardet             3.0.4
dbus-python         1.2.16
idna                2.8
numpy               1.20.3
pandas              1.2.4
pip                 21.1.2
plotly              4.14.3
pyarrow             2.0.0
PyGObject           3.36.0
python-apt          2.0.0+ubuntu0.20.4.5
python-dateutil     2.8.1
pytz                2021.1
requests            2.22.0
requests-unixsocket 0.2.0
retrying            1.3.3
scipy               1.6.3
setuptools          45.2.0
six                 1.14.0
urllib3             1.25.8
wheel               0.34.2
xmlrunner           1.7.7
```

**Comment by dongjoon-hyun** [comment]
Here is the PR, @xinrong-databricks and @HyukjinKwon 
- https://github.com/apache/spark/pull/32755

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:223]: Can we remove it back now, since it's fixed?

Comment (xinrong-databricks) [reply .github/workflows/build_and_test.yml]: Certainly, removed.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
I verified the GitHub Action result on PR manually.
Screen Shot 2021-06-03 at 2 23 45 PM

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @xinrong-databricks and @HyukjinKwon .

**Comment by xinrong-databricks** [comment]
Thanks @dongjoon-hyun !

**Pull request closed without merging**